Projects/Contracts/Sources/View/CoInsured/AlreadyExistingCoInsured/InsuredPeopleScreen.swift

```diff
@@ -0,0 +1,201 @@
+                        action: .openCoInsuredInput(
+                            isDeletion: true,
+                            name: coInsured.name,
+                            personalNumber: coInsured.SSN,
+                            title: L10n.contractRemoveCoinsuredConfirmation,
+                            contractId: contractId
+                        )
+                    )
+                )
+            }
+    }
+
+    var cancelButton: some View {
+        hButton.LargeButton(type: .ghost) {
+            store.send(.coInsuredNavigationAction(action: .dismissEditCoInsuredFlow))
+        } content: {
+            hText(L10n.generalCancelButton)
+        }
+        .padding(.horizontal, 16)
+    }
+
+    var confirmChangesView: some View {
+        VStack(spacing: 16) {
+            VStack(spacing: 2) {
+                HStack(spacing: 8) {
+                    hText(L10n.contractAddCoinsuredTotal)
+                    Spacer()
+
+                    if #available(iOS 16.0, *) {
+                        hText("129" + " " + L10n.paymentCurrencyOccurrence)
+                            .strikethrough()
+                            .foregroundColor(hTextColor.secondary)
+                    } else {
+                        hText("129" + " " + L10n.paymentCurrencyOccurrence)
+                            .foregroundColor(hTextColor.secondary)
+
+                    }
+                    hText("159" + " " + L10n.paymentCurrencyOccurrence)
+                }
+                hText(
+                    L10n.contractAddCoinsuredStartsFrom("2023-11-16".localDateToDate?.displayDateDDMMMYYYYFormat ?? ""),
+                    style: .footnote
+                )
+                .foregroundColor(hTextColor.secondary)
+                .frame(maxWidth: .infinity, alignment: .trailing)
+            }
+
+            LoadingButtonWithContent(ContractStore.self, .postCoInsured) {
+                /* TODO: SEND MUTATION */
+                store.send(.coInsuredNavigationAction(action: .openCoInsuredProcessScreen(showSuccess: true)))
+            } content: {
+                hText(L10n.contractAddCoinsuredConfirmChanges)
+            }
+        }
+        .padding(.horizontal, 16)
+    }
+}
+
+struct InsuredPeopleScreen_Previews: PreviewProvider {
+    static var previews: some View {
+        InsuredPeopleScreen(contractId: "")
+    }
+}
+
+class InsuredPeopleNewScreenModel: ObservableObject {
+    @Published var coInsuredAdded: [CoInsuredModel] = []
+    @Published var coInsuredDeleted: [CoInsuredModel] = []
+
+    var resetCoInsured: Void {
+        coInsuredAdded = []
+    }
+
+    func addCoInsured(name: String, personalNumber: String) {
+        coInsuredAdded.append(CoInsuredModel(name: name, SSN: personalNumber))
+    }
+
+    func removeCoInsured(name: String, personalNumber: String) {
+        let removedCoInsured = CoInsuredModel(name: name, SSN: personalNumber)
+        if coInsuredAdded.contains(removedCoInsured) {
```

Comment:
try like this:
 if let index = coInsuredAdded.firstIndex(of: removedCoInsured) {
 //logic for existing with index
 }